streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -152,6 +152,22 @@ public void initialize() {
         try {
             keyBytes = keySerializer.serialize(topic, headers, key);
             valBytes = valueSerializer.serialize(topic, headers, value);
+        } catch (final ClassCastException exception) {
```

Comment:
Moved this from `SinkNode` to here -- otherwise, we wrap the `ClassCastException` with a `StreamsException` and the `SinkNode` would not add the correct error message.